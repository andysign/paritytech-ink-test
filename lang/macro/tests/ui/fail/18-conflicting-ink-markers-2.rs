use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    #[ink(event)]
    #[ink(storage)] // We cannot have #[ink(storage)] if we already have #[ink(event)]
    struct Event {}

    impl Noop {
        #[ink(constructor)]
        fn new(&mut self) {}

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
