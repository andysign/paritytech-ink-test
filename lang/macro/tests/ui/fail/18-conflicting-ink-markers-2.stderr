error: invalid ink! attribute found for `#[ink(event)]` struct
 --> $DIR/18-conflicting-ink-markers-2.rs:9:10
  |
9 |     #[ink(storage)] // We cannot have #[ink(storage)] if we already have #[ink(event)]
  |          ^^^^^^^^^
